util/src/main/java/org/ethereum/beacon/util/ConsumerList.java

```diff
@@ -0,0 +1,125 @@
+package org.ethereum.beacon.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Iterators;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public final class ConsumerList<E> extends LinkedList<E> {
```

Comment:
You are extending `LinkedList` and using another `LinkedList` instance as `delegate`. Does it make sense here?